[doc/classes/int] fix description of operator '~'

umarcor opened the pull request:
The description of the `~` operation in class `int` is not correct. It states that a bitwise NOT operation results in `-int + 1`. In fact, due to how 2's complement works, the operation results in `-(int +1)`. This PR updates the description and fixes the examples in the codeblock.

umarcor committed:
[doc/classes/int] fix description of operator '~'

AThousandShips commented:
I believe it's `[url=address]text[/url]`

umarcor commented:
@AThousandShips thanks! :heart: I updated it with the fixed syntax.

AThousandShips commented:
You need to escape the arrows for html, `-&gt;`, I'd recommend running `--doctool` locally to get these things

MewPurPur commented:
Related: #70046. I didn't catch this mistake in particular, but there are other mistakes the PR fixes.

umarcor commented:
> You need to escape the arrows for html, `-&gt;`, I'd recommend running `--doctool` locally to get these things

I tried running `--doctool` with the same release version of godot I use for developing games. Unfortunately that changed the version in many of the files in the doc subdir. Hence, I had to build godot from sources. That took some time. Nevertheless, I must say I was very gratefully surprised by how easy it was.

I now changed `->` to `-&gt;` through doctool and updated this PR. I wanted to do it with the tool rather than fixing it manually.

umarcor commented:
> As for this PR, I'm not too fond of the examples. I don't think we need that many of them, and the binary representations are also weird. Having only 6 bits to represent complement numbers is awkward, moreover I'm not sure if we should concern users with complement numbers in the first place.

I reduced the examples to just one positive and one negative.

With regard to the word-length of the bit representation, since the sign is extended we can use any number of bits >=4. I'm good with using just 4 or 8, if 6 feels uncommon.

With regard to concerning users with complement numbers, that's fundamental for understanding logical operations on integers. The mistake this PR fixes would be avoided if the binary representation had been written/reviewed. I find it possitive (not concerning) that the reference might trigger the curiosity of users expecting the more human readable but uncommon sign+magnitude or ones' complement.

AThousandShips commented:
My bad I should have been clear that it had to be the built version!

YuriSizov commented on doc/classes/int.xml:
```diff
@@ -397,10 +397,11 @@
 			<return type="int" />
 			<description>
-				Returns the result of bitwise [code]NOT[/code] operation for the integer. It's effectively equal to [code]-int + 1[/code].
+				Returns the result of bitwise [code]NOT[/code] operation for the integer.
+				Due to [url=https://en.wikipedia.org/wiki/Two%27s_complement/]2's complement[/url], it's effectively equal to [code]-(int + 1)[/code].
```
There is no need to create a new paragraph here.

YuriSizov reviewed (approved):
Looks good, except for a formatting issue.

umarcor commented:
I updated the comments to use the same literal syntax as in https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#literals (six bit prefixed with 0b). I also replace `->` with `=` (in a mathematical sense, not as an assignment).

YuriSizov commented:
Superseded by https://github.com/godotengine/godot/pull/70046. Thanks for your contribution nevertheless!
